src/lib/int.ts

```diff
@@ -1,15 +1,18 @@
-import { Circuit, Field } from '../snarky';
+import { Circuit, Field, Types } from '../snarky';
 import { CircuitValue, prop } from './circuit_value';
 
 export { UInt32, UInt64, Int64 };
 
-class UInt64 extends CircuitValue {
+class UInt64 extends CircuitValue implements Types.UInt64 {
   @prop value: Field;
   static NUM_BITS = 64;
+  type = 'UInt64' as const;
 
   constructor(value: Field) {
     super();
     this.value = value;
+    // TODO type breaks Circuit.if for some reason
+    delete (this as any).type;
```

Comment:
we actually can't keep the `.type` property on the class instance.. it's just for typescript